stingray/lightcurve.py

```diff
@@ -185,3 +332,266 @@ def rebin_lightcurve(self, dt_new, method='sum'):
 
         lc_new = Lightcurve(bin_time, bin_counts)
         return lc_new
+
+    def join(self, other):
+        """
+        Join two lightcurves into a single object.
+
+        The new Lightcurve object will contain time stamps from both the
+        objects. The count per bin in the resulting object will be the
+        individual count per bin, or the average in case of overlapping
+        time arrays of both lightcurve objects.
+
+        Note : Time array of both lightcurves should not overlap each other.
+
+        Parameters
+        ----------
+        other : Lightcurve object
+            The other Lightcurve object which is supposed to be joined with.
+
+        Returns
+        -------
+        lc_new : Lightcurve object
+            The resulting lightcurve object.
+
+        Example
+        -------
+        >>> time1 = [5, 10, 15]
+        >>> count1 = [300, 100, 400]
+        >>> time2 = [20, 25, 30]
+        >>> count2 = [600, 1200, 800]
+        >>> lc1 = Lightcurve(time1, count1)
+        >>> lc2 = Lightcurve(time2, count2)
+        >>> lc = lc1.join(lc2)
+        >>> lc.time
+        array([ 5, 10, 15, 20, 25, 30])
+        >>> lc.counts
+        array([ 300,  100,  400,  600, 1200,  800])
+        """
+        if self.dt != other.dt:
+            utils.simon("The bin widths of both the lightcurves are not "
+                        "same.")
+
+        if self.tstart <= other.tstart:
+            new_time = np.unique(np.concatenate([self.time, other.time]))
+        else:
+            new_time = np.unique(np.concatenate([other.time, self.time]))
+
+        if len(new_time) != len(self.time) + len(other.time):
+            utils.simon("Both the lightcurves have overlapping time ranges. "
+                        "For the common time range, resulting count will be "
+                        "the average of the counts in both lightcurves. If "
+                        "you wish to sum, use `lc_sum = lc1 + lc2`.")
+
```

Comment:
"The two light curves have overlapping time ranges.
In the common time range, THE resulting counts will be the average of the counts in the two light curves. (...)"